src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,105 @@
+package duke.task;
+
+/**
+ * The Task Class is a representation of a task that Duke is keeping track of.
+ *
+ * It contains information relating to the task:
+ * - description
+ * - isDone
+ *
+ * @author Benedict Chua
+ */
+public abstract class Task {
+    private String description;
+    private boolean isDone;
+
+    /**
+     * Constructor for a Task.
+     *
+     * @param description The name of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Convert the Task into a suitable String for saving into file.
+     * To be overridden by child classes to call formatString in a suitable way.
+     *
+     * @return String of the formatted Task.
+     */
+    public abstract String saveAsString();
```

Comment:
Can consider changing the name of this method to `convertToString()`

Reply:
saveAsString seems to be ok here

Reply:
I agree that convertToString might be a better name since I am returning a String and not really saving it (void function). Thanks for the suggestion! 😃